### PR TITLE
enable VM resource quota for AWS provider

### LINKED ISSUE #6459: Add resource quota support for AWS provider

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: enable VM resource quota for AWS provider

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6459

```release-note
enable VM resource quota for AWS provider
```

### CODE REVIEW

File: pkg/handler/common/provider/aws.go [hunk 1]
```diff
@@ -222,36 +227,64 @@ func SetDefaultSubnet(machineDeployments *clusterv1alpha1.MachineDeploymentList,
...
+func filterAWSByQuota(instances apiv1.AWSSizeList, quota kubermaticv1.MachineDeploymentVMResourceQuota) apiv1.AWSSizeList {
+	filteredRecords := apiv1.AWSSizeList{}
+
+	// Range over the records and apply all the filters to each record.
+	// If the record passes all the filters, add it to the final slice.
+	for _, r := range instances {
+		keep := true
+
+		// Filter too expensive instance types (>1$ per hour) if GPU not enabled
+		if !quota.EnableGPU && r.Price > 0 {
```
Comment (lsviben) [inline pkg/handler/common/provider/aws.go]: `r.Price > 1` should be here.

BTW should we still filter these (>1$ per hour), I am not sure if users are aware of this restriction. Or maybe we could add another filter for price (in a different ticket).

**Review by lsviben — CHANGES_REQUESTED** [review]

File: pkg/handler/common/provider/aws.go (hunk 1 above)
Comment (zreigz) [reply pkg/handler/common/provider/aws.go]: There was a request for this some time ago, so I decided to leave this.

**Review by xrstf — APPROVED** [review]
The LGTM label was removed when I added some debugging code to fix our CI cluster. Now that the debugging code is gone, this PR can be LGTM again.

**Pull request merged**